### ISSUE TITLE
No sleep option in power menu

### ISSUE THREAD

**Issue by ivlis** [pop-os/cosmic-epoch#859]
I'm not sure if that's by design or it's a bug, but I don't see a sleep option in the power menu:

![screenshot-2024-08-30-13-07-38](https://github.com/user-attachments/assets/fa58350f-204f-4eca-8434-2af29d20b529)

The settings also have no option to configure sleep for the laptop lid state.

**Comment by git-f0x** [comment]
Suspend is sleep.

**Comment by ivlis** [comment]
Okay, true; that means hibernate is missing :)

**Comment by git-f0x** [comment]
I think the general trend in distros is to remove hibernate, because it can have issues.

**Comment by locomot1f** [comment]
this is also mentioned in Issue: Feature Request - #537 
I'm also looking forward to these options hopefully making it to this DE.

@git-f0x - Is this considered a distro? I was under the impression that this is a DE. These features exists on in POP! OS.

**Comment by git-f0x** [comment]
No, it is a DE. But I've seen some distros (e.g. Fedora) not having a hibernate option, due to issues.
It could probably be conditionally added if it's a really wanted feature (though sleep and shut down are probably enough for most people).

**Comment by locomot1f** [comment]
i had to do my research on the difference between Hibernate and Suspend.

**Comment by Department-stack** [comment]
IMO hibernate really isn't worth it. Sleep/suspend has lasted me weeks on a full battery

**Comment by iamkartiknayak** [comment]
Hibernate is worth a shot if it can be implemented. Using hibernate battery will also last longer compared to sleep. During sleep if your battery falls to zero your entire work that was open on your system will be destroyed if not saved. Hibernate allow saving the state of apps so no matter if battery is emptied, data will be available on next boot as it was left before hibernating the system.

**Comment by Department-stack** [comment]
> If implemented it can also be used to prevent data loss when during sleep if battery falls below 5% system will hibernate before state of system is killed

I haven't thought about that

**Comment by iamkartiknayak** [comment]
That's how it works in Windows. We can set a battery percent threshold, below which system will hibernate

**Comment by feikedonia** [comment]
Hibernation is handy when you are multi-tasking, when for example researching a project for which you have different apps open. Suddenly you have to go, and you don't know when you are back: hibernate your system, doesn't use power. Suspend uses power, so when the battery runs out, you loose your workflow.

**Comment by mohamed591195** [comment]
I'm also tired of the endless workarounds and articles required just to enable hibernation. Even after all that hassle, it still doesn't work properly. That's why I fully support adding this feature—it’s so underrated.

**Comment by feikedonia** [comment]
I've issued a feature request against pop-os/iso to make installing hibernation on Pop!_OS as seemless as possible.
https://github.com/pop-os/cosmic-epoch/issues/859

**Comment by raphmim** [comment]
The lack of hibernate option is one of the big reasons I've moved back to Windows after each attempt at Linux desktop. I'm really hoping to stick to Fedora COSMIC this time, but hibernate would definitely make adoption easier for people coming from Mac and Windows.

**Comment by feikedonia** [comment]
If you have a big enough swap-partition, you can set a keyboard-short-cut for the command `systemctl hibernate`.

_Settings → Input-devices → Keyboard → Keyboard-short-cuts → Custom_

![Image](https://github.com/user-attachments/assets/4ffc20a5-2965-4e2b-bc7c-7c1f35322efd)

**Comment by loganmancuso** [comment]
> If you have a big enough swap-partition, you can set a keyboard-short-cut for the command `systemctl hibernate`.
> 
> _Settings → Input-devices → Keyboard → Keyboard-short-cuts → Custom_
> 
> ![Image](https://github.com/user-attachments/assets/4ffc20a5-2965-4e2b-bc7c-7c1f35322efd)

thanks for the idea, i think a keyboard shortcut is the best option until it can be added to the DE. 

I am running cosmic alpha and followed the instructions here and Hibernate works great. https://support.system76.com/articles/enable-hibernation/

**Comment by dairymilkbatman** [comment]
There are some linux systems which have problems with suspend, it isn't just the other way around.

**Comment by phoekjelle** [comment]
> There are some linux systems which have problems with suspend, it isn't just the other way around.

Yes, but that's the user's decission, aint it? You could just provide a warning. 

That's my opinion.

**Comment by FelixBrakel** [comment]
Sleep and hibernate are two different features. One doesn't replace the other. And then there is also hybrid sleep, which requires both.

In any case the actual implementation of hibernation is done on a system level and requires you to have a swap file or partition set up. I think what we're really asking for here is a button in the power menu applet to trigger the command.

**Comment by 31337-4554551n** [comment]
On this note, are there any clear guides on moving the swap partition to inside the LVM?
I've tried a few things using kde partition manager as gui, but nothing really works, and at best it fails to boot, at worst it complains about errors and that the thing has been resized.

I am all about that hibernate, but hibernating on an unencrypted partition doesn't seem great.

Would be good to have an option-or at least simple steps, to remove the default swap partition, expand the lvm, shrink the data volume in the lvm, and create/activate a new swap partition in the luks volume.
The last bit is where I keep having issues.

**Comment by 31337-4554551n** [comment]
Is there any more info/updates on this?
Even without a hibernate button, at least a way to set up an encrypted swap partition within the LVM.
Ideally this would be an option in the installer, but at least a guide of how to manually make it after the fact would be useful.

**Comment by toadslop** [comment]
Just to add some more context from my own use case.

## My problem

I have a laptop with a firmware bug that causes Linux to fail to poweroff using the normal shutdown process, but the hibernate path works correctly. In order to poweroff my machine, I need to use hibernate. I can do this via the command line easy enough, but it would be nice to be able to configure cosmic for this case so the UI works with the machine seemlessly.

I would consider this a low priority feature given that hibernate can be achieved via the command line.

For reference, I currently run a patched version of cosmic-osd on my system that uses hibernate rather than shutdown; I would prefer to be able to configure this rather than patch the software.

## Possible Solutions (simple to complex)

Solutions for my case are more unique since ideally I don't want the UI to provide a button for something that does not work. Rather than simply providing a hibernate button, my use case involves not exposing broken behavior via the UI.

### Configuration Shutdodwn Button Behavior

Allow configuring the behavior of the existing shutdown button (in other words, via a configuration file, a user can specify to use 'hibernate' or 'shutdown').

#### Pros

- simple to implement

#### Cons

- Can't have both hibernate and shutdown available at the same time

### Provide Both Buttons with Hide/Disable configuration

Expose both a shutdown and hibernate button, but via the configuration file allow users to either hide or disable one of them.

#### Pros

- Relatively simple to implement
- More flexibility to the user
- Allows both buttons to be present

#### Cons

- Slightly more complex than the first option

## What I can Do

If I can get confirmation that the PR would be merged, I'd be happy to implement one of the above options.

**Comment by sriharshachilakapati** [comment]
I'm trying to enable Hibernate, but found out that hibernation will not be supported if you are using Secure Boot.
